src/ssl.c

```diff
@@ -10420,6 +10802,10 @@ int wolfSSL_Cleanup(void)
 
     WOLFSSL_ENTER("wolfSSL_Cleanup");
 
+#if defined(WOLFSSL_SYS_CRYPTO_POLICY)
+    wolfSSL_crypto_policy_disable();
```

Comment:
I think the policy disable should only happen on a full free. After the initRefCount is 0.